### ISSUE TITLE
NullPointerException in DataCodec.encode in CreateDestroyICacheTest simulator test [HZ-1187]

### ISSUE THREAD

**Issue by olukas** [hazelcast/hazelcast#20764]
Simulator run in lab for 5.1 finished with NPE for `CreateDestroyICacheTest` after 17hours. 35 clients failed with this NPE:
```
Failure[
   message='Worked ran into an unhandled exception'
   type=WORKER_EXCEPTION
   timestamp=1645254721506
   duration=61696337 ms
   workerAddress=A10_W28
   agentAddress=10.212.1.110
   workerId=A10_W28-10.212.1.110-javaclient
   test=TestCase{
        id=CacheCreateDestroy,
        class=com.hazelcast.simulator.tests.icache.CreateDestroyICacheTest,
        closeCacheProb=0.025,
        createCacheProb=0.15,
        destroyCacheProb=0.025,
        putCacheProb=0.8,
        threadCount=3
    }
   cause=java.lang.NullPointerException
        at com.hazelcast.client.impl.protocol.codec.builtin.DataCodec.encode(DataCodec.java:32)
        at com.hazelcast.client.impl.protocol.codec.custom.CacheConfigHolderCodec.encode(CacheConfigHolderCodec.java:66)
        at com.hazelcast.client.impl.protocol.codec.builtin.CodecUtil.encodeNullable(CodecUtil.java:50)
        at com.hazelcast.client.impl.protocol.codec.CacheGetConfigCodec.encodeResponse(CacheGetConfigCodec.java:94)
        at com.hazelcast.client.impl.protocol.task.cache.CacheGetConfigMessageTask.encodeResponse(CacheGetConfigMessageTask.java:85)
        at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.sendResponse(AbstractMessageTask.java:243)
        at com.hazelcast.client.impl.protocol.task.AbstractAsyncMessageTask.sendResponseOrHandleFailure(AbstractAsyncMessageTask.java:79)
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:760)
        at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:736)
        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:442)
        at com.hazelcast.internal.util.ConcurrencyUtil$1.execute(ConcurrencyUtil.java:39)
        at java.util.concurrent.CompletableFuture$UniCompletion.claim(CompletableFuture.java:529)
        at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:751)
        at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:736)
        at java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:781)
        at java.util.concurrent.CompletableFuture.whenCompleteAsync(CompletableFuture.java:2150)
        at com.hazelcast.client.impl.protocol.task.AbstractAsyncMessageTask.processMessage(AbstractAsyncMessageTask.java:73)
        at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.initializeAndProcessMessage(AbstractMessageTask.java:153)
        at com.hazelcast.client.impl.protocol.task.AbstractMessageTask.run(AbstractMessageTask.java:116)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
        at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
        at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)
        at ------ submitted from ------.()
        at com.hazelcast.internal.util.ExceptionUtil.cloneExceptionWithFixedAsyncStackTrace(ExceptionUtil.java:336)
        at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.returnOrThrowWithGetConventions(InvocationFuture.java:112)
        at com.hazelcast.client.impl.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:95)
        at com.hazelcast.client.impl.spi.impl.ClientInvocationFuture.resolveAndThrowIfException(ClientInvocationFuture.java:40)
        at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:617)
        at com.hazelcast.client.cache.impl.ClientCacheHelper.getCacheConfig(ClientCacheHelper.java:66)
        at com.hazelcast.client.cache.impl.HazelcastClientCacheManager.getCacheConfig(HazelcastClientCacheManager.java:105)
        at com.hazelcast.client.cache.impl.HazelcastClientCacheManager.findCacheConfig(HazelcastClientCacheManager.java:132)
        at com.hazelcast.cache.impl.AbstractHazelcastCacheManager.getCacheUnchecked(AbstractHazelcastCacheManager.java:228)
        at com.hazelcast.cache.impl.AbstractHazelcastCacheManager.getCache(AbstractHazelcastCacheManager.java:210)
        at com.hazelcast.cache.impl.AbstractHazelcastCacheManager.getCache(AbstractHazelcastCacheManager.java:65)
        at com.hazelcast.simulator.tests.icache.CreateDestroyICacheTest.putCache(CreateDestroyICacheTest.java:64)
        at CreateDestroyICacheTestRunnerCacheCreateDestroy.timeStepLoop(CreateDestroyICacheTestRunnerCacheCreateDestroy.java:46)
        at com.hazelcast.simulator.worker.testcontainer.TimeStepRunner.run(TimeStepRunner.java:107)
        at java.lang.Thread.run(Thread.java:745)
        at com.hazelcast.simulator.utils.ThreadSpawner$ReportExceptionThread.run(ThreadSpawner.java:179)
]
```

**Comment by olukas** [comment]
Reopening since the same happened in Simulator run for 5.1.1 after 41 hours.

**Comment by olukas** [comment]
It failed also in simulator run for 5.0.3 after 35 hours.

**Comment by degerhz** [comment]
Fixed by #22844

**Comment by Patras3** [comment]
Reopening, the same failure happened again for the Simulator run during releasing of 5.2.3 version after almost 24h.
@vbekiaris could you please check if the fix candidate can be merged to all supported 5.x branches?
